test/integration/discordactions.test.js

```diff
@@ -269,4 +273,65 @@ describe("Discord actions", function () {
         });
     });
   });
+
+  describe("PUT /discord-actions/group-onboarding-31d-plus", function () {
+    beforeEach(async function () {
+      userData[0] = {
+        ...userData[0],
+        discordId: "123456789098765432",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+      userData[1] = {
+        ...userData[1],
+        discordId: "12345678909867666",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+      userData[2] = {
+        ...userData[2],
+        discordId: "123456",
+        discordJoinedAt: "2023-07-31T16:57:53.894000+00:00",
+        roles: { archived: false, in_discord: true },
+      };
+      const userId1 = await addUser(userData[0]);
+      const userId2 = await addUser(userData[1]);
+      const userId3 = await addUser(userData[2]);
+
+      await updateUserStatus(userId1, generateUserStatusData("ONBOARDING", new Date(), new Date()));
+      await updateUserStatus(userId2, generateUserStatusData("ONBOARDING", new Date(), new Date()));
+      await updateUserStatus(userId3, generateUserStatusData("ONBOARDING", new Date(), new Date()));
+
+      const addRolesPromises = [discordRoleModel.add(groupOnboarding31dPlus)];
+      await Promise.all(addRolesPromises);
+
+      fetchStub.returns(
+        Promise.resolve({
+          status: 200,
+          json: () => Promise.resolve(getDiscordMembers),
+        })
+      );
+    });
+    afterEach(async function () {
+      sinon.restore();
+      await cleanDb();
+    });
+
+    it("should update role for onboarding users with 31 days completed", function (done) {
+      chai
+        .request(app)
+        .put(`/discord-actions/group-onboarding-31d-plus`)
+        .set("Cookie", `${cookieName}=${superUserAuthToken}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(201);
+          expect(res.body.message).to.be.equal("All Users with 31 Days Plus Onboarding are updated successfully.");
+          expect(res.body.totalOnboardingUsers31DaysCompleted.count).to.be.equal(3);
+          expect(res.body.totalOnboarding31dPlusRoleApplied.count).to.be.equal(3);
```

Comment:
Since this API also removes the role, please consider adding tests for the same.

Reply:
Done now. @Ajeyakrishna-k Please check.